services/src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutTransactionService.java

```diff
@@ -163,10 +179,9 @@ public PaymentResponse capture(PaymentRequest paymentRequest) {
 
     @Override
     public PaymentResponse authorizeAndCapture(PaymentRequest paymentRequest) {
-        PaymentResponse paymentResponse =
-                new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
-                        PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT)
-                                .transactionType(DefaultTransactionTypes.AUTHORIZE_AND_CAPTURE);
+        PaymentResponse paymentResponse = paymentResponseUtil.buildPaymentResponse(paymentRequest,
+                PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT,
+                DefaultTransactionTypes.AUTHORIZE_AND_CAPTURE);
```

Comment:
We should move the `paymentResponseUtil.buildPaymentResponse(...)` call inside of the `try` block & we should catch `Exception` instead of `PaymentException`, to guarantee that this method will always return a PaymentResponse